users/views.py

```diff
@@ -1,3 +1,51 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.views.generic import TemplateView
+from django.views import View
+from django.contrib.auth import login, logout
+from django.contrib.auth.mixins import LoginRequiredMixin
 
-# Create your views here.
+from .forms import LoginForm
+
+
+class UserLogin(TemplateView):
+    """ User login account
+    """
+    template_name = 'auth/login.html'
+
+    def get(self, *args, **kwargs):
+        """ Renders the login form
+        """
+        form = LoginForm()
+        return render(self.request, self.template_name, {'form':form})
+
+    def post(self, *args, **kwargs):
+        """ submits the data
+        """
+        form = LoginForm(self.request.POST)
+        if form.is_valid():
+            login(self.request, form.user)
+            return redirect('Dashboard')
+
+        return render(self.request, self.template_name, {'form':form})
+
+
+class Dashboard(LoginRequiredMixin, TemplateView):
+    """ Displays the dashboard page and used LoginRequiredMixin 
+        to check the user if logged in.
+    """
+    login_url = '/login/'
+    redirect_field_name = 'redirect_to'
```

Comment:
what is this?

Reply:
They are for the login required mixins sir.